javascript/ql/src/semmle/javascript/frameworks/jQuery.qll

```diff
@@ -496,6 +496,40 @@ module JQuery {
         hasUnderlyingType("jQuery")
       }
     }
+
+    /**
+     * Gets a node that is written to `$.fn[something]`.
+     * JQuery plugins are registered this way.
+     */
+    private DataFlow::Node getAFnWrite() {
```

Comment:
Except for the requirement of the known plugin name, this seems to do exactly what you need: https://github.com/Semmle/ql/blob/801ce89c67c08e56beb0ebf88042d5b83ca4f1db/javascript/ql/src/semmle/javascript/security/dataflow/UnsafeJQueryPluginCustomizations.qll#L55

Reply:
It is close enough that I can reuse it.